veracruz-client/src/admin_client.rs

```diff
@@ -0,0 +1,145 @@
+//! The Veracruz admin client
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use crate::error::VeracruzClientError;
+use std::convert::identity;
+use std::num::NonZeroU32;
+use std::time::Duration;
+use veracruz_utils::policy::policy::Policy;
+use serde::{Serialize, Deserialize};
+
+
+/// Provides VeracruzAdminClient, a client API for administrative actions
+/// that don't require a policy
+///
+/// For interaction with the actual enclave see VeracruzClient
+///
+#[derive(Debug)]
+pub struct VeracruzAdminClient {
+    url: String,
+}
+
+/// Status of enclaves running on a Veracruz server, returned by
+/// enclave_list
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct VeracruzEnclaveStatus {
+    pub policy_hash: String,
+    pub id: u32,
+    pub uptime: Duration,
+}
+
+impl VeracruzAdminClient {
+    /// Create a new VeracruzAdminClient to talk to the given server
+    pub fn new(url: &str) -> Self {
```

Comment:
Should this parameter be `String` rather than taking a slice and allocating?

Reply:
See https://github.com/veracruz-project/veracruz/pull/242#discussion_r821761233